crates/swc_ecma_minifier/src/compress/optimize/iife.rs

```diff
@@ -301,7 +301,7 @@ where
     /// If a parameter is not used, we can ignore return value of the
     /// corresponding argument.
     pub(super) fn ignore_unused_args_of_iife(&mut self, e: &mut CallExpr) {
-        if !self.options.unused {
+        if !self.options.unused && !self.options.reduce_vars {
```

Comment:
I changed this because of https://github.com/swc-project/swc/blob/a0b0fd3e0d8a1084a0399e8b0345d66cfe835b0f/crates/swc_ecma_minifier/tests/terser/compress/drop_unused/chained_3/input.js#L1-L8